users-auth/auth-router.js

```diff
@@ -0,0 +1,72 @@
+const router = require('express').Router();
+const bcrypt = require("bcryptjs")
+const jwt = require("jsonwebtoken")
+const Users = require("./auth-model")
+
+router.post('/register', async (req, res, next) => {
+	try {
+		const { username, password, phoneNumber} = req.body
+		const user = await Users.findBy({ username }).first()
+
+		if (user) {
+			return res.status(409).json({
+				message: "Username is already taken",
+			})
+		}
+
+		const newUser = await Users.addUser({
+			username,
+            password: await bcrypt.hash(password, 8),
+            phoneNumber,
+			
+		})
+
+		res.status(201).json({message: `Welcome, ${newUser.username}!`})
+	} catch(err) {
+		next(err)
+	}
+});
+
+
+router.post('/login', async (req, res, next) => {
+	try {
+		const { username, password, id } = req.body
+		const user = await Users.findBy({ username }).first()
```

Comment:
Username may not exist in the body and might throw from knex an error if undefined.